app/models/project_group.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: timeline_entries
+#
+#  agent_id          :integer
+#  completed_date    :date
+#  created_at        :datetime         not null
+#  date_change_count :integer          default(0), not null
+#  finalized_at      :datetime
+#  id                :integer          not null, primary key
+#  is_finalized      :boolean
+#  original_date     :date
+#  parent_id         :integer
+#  project_id        :integer
+#  project_type      :string
+#  scheduled_date    :date
+#  step_type_value   :string
+#  type              :string           not null
+#  updated_at        :datetime         not null
+#
+# Indexes
+#
+#  index_timeline_entries_on_agent_id                     (agent_id)
+#  index_timeline_entries_on_project_type_and_project_id  (project_type,project_id)
+#
+# Foreign Keys
+#
+#  fk_rails_a9dc5eceeb  (agent_id => people.id)
+#  fk_rails_d21c3b610d  (parent_id => timeline_entries.id)
+#
+
+require 'chronic'
+class ProjectGroup < TimelineEntry
+  class DestroyWithChildrenError < StandardError; end
+
+  # Prepend required to work with has_closure_tree,
+  # otherwise children are deleted before we even get here.
+  before_destroy :validate_no_children, prepend: true
```

Comment:
Ugh, refactored this to a callback instead. Sunk a ton of time into it, due to a bug in closure tree (at least unexpected functionality).

If you call destroy, by the time your callback is fired, closure tree has deleted the children... [lib/closure_tree/hierarchy_maintenance.rb](https://github.com/mceachen/closure_tree/blob/8089f887540aad58f3ae828036e26614427df241/lib/closure_tree/hierarchy_maintenance.rb#L52 )